BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Public/Span.cs

```diff
@@ -167,5 +152,57 @@ public void UpdateSamplingProbability(double value)
             }
         }
 
+
+        internal void SetAttributeInternal(string key, int value) => SetAttributeWithoutChecks(key, value);
+        internal void SetAttributeInternal(string key, string value) => SetAttributeWithoutChecks(key, value);
+        internal void SetAttributeInternal(string key, double value) => SetAttributeWithoutChecks(key, value);
+        internal void SetAttributeInternal(string key, bool value) => SetAttributeWithoutChecks(key, value);
+        private void SetAttributeWithoutChecks(string key, object value)
+        {
+            if (value == null)
+            {
+                if (_attributes.ContainsKey(key))
+                {
+                    _attributes.Remove(key);
+                }
+                return;
```

Comment:
Any reason we can't skip the `ContainsKey` check? `Remove` shouldn't produce an error if the key wasn't found.

```suggestion
                _attributes.Remove(key);
                return;
```